examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -164,23 +149,30 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
                         purchaseParamsBuilder.isPersonalizedPrice(isPersonalizedPrice)
                     }
 
-                    Purchases.sharedInstance.purchaseWith(
-                        purchaseParamsBuilder.build(),
-                        purchaseErrorCallback,
-                        successfulPurchaseCallback,
-                    )
+                    purchase(purchaseParamsBuilder.build())
                 }
             }
         } else {
             if (isPersonalizedPrice) {
                 purchaseParamsBuilder.isPersonalizedPrice(isPersonalizedPrice)
             }
+            purchase(purchaseParamsBuilder.build())
+        }
+    }
 
-            Purchases.sharedInstance.purchaseWith(
-                purchaseParamsBuilder.build(),
-                purchaseErrorCallback,
-                successfulPurchaseCallback,
-            )
+    private fun purchase(params: PurchaseParams) {
+        scope.launch {
+            try {
+                val (storeTransaction, _) = Purchases.sharedInstance.awaitPurchase(params)
+                toggleLoadingIndicator(false)
+                handleSuccessfulPurchase(storeTransaction.orderId)
+            } catch (exception: PurchasesTransactionException) {
+                toggleLoadingIndicator(false)
+                showUserError(
```

Comment:
We shouldn't show user error if it's a cancellation I think?

Reply:
whoops, bad refactor, updating